views/challenge.py

```diff
@@ -60,8 +69,16 @@ class TypeCheckResult:
 
 
 class ChallengeManager:
-    def __init__(self):
-        self.challenges: dict[ChallengeKey, Challenge] = self._load_challenges()
+    """The manager for challenges.
+
+    :param root_dir: The root directory that contains the files of challenges,
+        use "{PROJECT_ROOT}/challenges" by default.
```

Comment:
This feels redundant and can be removed.

Reply:
removed.